File: files/200-299/247/CH-247.R
library(tidyverse)
library(readxl)

path = "files/200-299/247/CH-247 Date Calculation.xlsx"
test = read_excel(path, range = "B2:B14") %>%
  mutate(Dates = as.Date(Dates))

result = data.frame(
  date = seq(as.Date("2025-01-01"), as.Date("2025-12-01"), by = "month")
) %>%
  mutate(
    Dates = map(date, ~ date + (8 - wday(date, week_start = 1)) %% 7),
    .by = "date"
  ) %>%
  select(Dates) %>%
  unnest(Dates)

all.equal(test$Dates, result$Dates)
# [1] TRUE

File: files/200-299/247/CH-247.py
from datetime import date, timedelta
import pandas as pd

path = "200-299/247/CH-247 Date Calculation.xlsx"
test = pd.read_excel(path, usecols="B", skiprows=1, nrows=12)

first_mondays = [
    d + timedelta((7 - d.weekday()) % 7)
    for d in [date(2025, m, 1) for m in range(1, 13)]
]
first_mondays = pd.to_datetime(first_mondays)
first_mondays = pd.DataFrame(first_mondays, columns=["Dates"])

print(test.equals(first_mondays))
# True